### PR TITLE
[8/n][dagster-airbyte] Implement load_airbyte_cloud_asset_specs

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

This PR implements the `load_airbyte_cloud_asset_specs` function:

```python
from dagster_airbyte import AirbyteCloudWorkspace, load_airbyte_cloud_asset_specs

import dagster as dg

airbyte_cloud_workspace = AirbyteCloudWorkspace(
    workspace_id=dg.EnvVar("AIRBYTE_CLOUD_WORKSPACE_ID"),
    client_id=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_ID"),
    client_secret=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_SECRET"),
)

airbyte_cloud_specs = load_airbyte_cloud_asset_specs(airbyte_cloud_workspace)
defs = dg.Definitions(assets=airbyte_cloud_specs, resources={"airbyte": airbyte_cloud_workspace}
```

## How I Tested These Changes

Additional unit test

## Changelog

[dagster-airbyte] The `load_airbyte_cloud_asset_specs` function is added. It can be used with the `AirbyteCloudWorkspace` resource and `DagsterAirbyteTranslator` translator to load your Airbyte Cloud connection streams as external assets in Dagster.

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26269?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26269** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26269?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26269?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26264** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26264?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26257** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26257?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26253** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26253?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26242** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26242?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26241** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26241?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26205** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26205?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26204** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26204?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py [hunk 1]
```diff
@@ -1059,3 +1070,70 @@ def fetch_airbyte_workspace_data(
...
+
+            airbyte_cloud_workspace = AirbyteCloudWorkspace(
+                workspace_id=dg.EnvVar("AIRBYTE_CLOUD_WORKSPACE_ID"),
+                client_id=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_ID"),
+                client_secret=dg.EnvVar("AIRBYTE_CLOUD_CLIENT_SECRET"),
+            )
+
+
+            airbyte_cloud_specs = load_airbyte_cloud_asset_specs(airbyte_cloud_workspace)
+            defs = dg.Definitions(assets=airbyte_cloud_specs, resources={"airbyte": airbyte_cloud_workspace}
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:1107]: you don't need the resource just to load asset specs right

File: python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py [hunk 2]
```diff
@@ -20,3 +22,27 @@ def test_fetch_airbyte_workspace_data(
...
+            client_id=EnvVar("AIRBYTE_CLIENT_ID"),
+            client_secret=EnvVar("AIRBYTE_CLIENT_SECRET"),
+        )
+
+        all_assets = load_airbyte_cloud_asset_specs(resource)
+        all_assets_keys = [asset.key for asset in all_assets]
+
+        # 4 tables for the connector
+        assert len(all_assets) == 1
+        assert len(all_assets_keys) == 1
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py:44]: this comment seems completely unrelated?

File: python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py [hunk 3]
```diff
@@ -20,3 +22,27 @@ def test_fetch_airbyte_workspace_data(
...
+
+        # Sanity check outputs, translator tests cover details here
+        first_asset_key = next(key for key in all_assets_keys)
+        assert first_asset_key.path == ["test_prefix_test_stream"]
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py:48]: name the actual tests in question please

File: python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py [hunk 4]
```diff
@@ -20,3 +22,27 @@ def test_fetch_airbyte_workspace_data(
...
+
+        all_assets = load_airbyte_cloud_asset_specs(resource)
+        all_assets_keys = [asset.key for asset in all_assets]
+
+        # 4 tables for the connector
+        assert len(all_assets) == 1
+        assert len(all_assets_keys) == 1
+
+        # Sanity check outputs, translator tests cover details here
+        first_asset_key = next(key for key in all_assets_keys)
```
Comment (dpeng817) [inline python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py:47]: should just call this `the_asset_key` since we expect one

**Review by dpeng817 — APPROVED** [review]

File: python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py (hunk 1 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py:1103]: Right, good catch

File: python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py (hunk 2 above)
Comment (maximearmstrong) [reply python_modules/libraries/dagster-airbyte/dagster_airbyte_tests/experimental/test_asset_specs.py:44]: Indeed - will fix
